components/renderers/controls/array/array-renderer.ts

```diff
@@ -120,10 +120,14 @@ class ArrayRenderer implements JSONForms.IRenderer {
             if (supportsSubmit) {
                 containerDescription['submitElement'] = submitElement;
                 containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
+                if (data == undefined) {
+                    containerDescription["instance"][containerDescription['path']] = [];
+                    data = containerDescription["instance"][containerDescription['path']];
+                }
                 containerDescription['submitCallback'] = () => data.push(_.clone(submitElement));
             }
             containerDescription['generateControlDescriptions'] = (data) =>
-                this.generateControlDescriptions(items, schemaPath, containerDescription['elements'], data.length, services);
+                this.generateControlDescriptions(items, schemaPath, containerDescription['elements'],  data.length, services);
```

Comment:
I fully agree, this is already tracked via https://github.com/eclipsesource/jsonforms/issues/105